### PR TITLE
Rubocop: resolve CI breaking issues

### PR DESCRIPTION
**Pull request by orien**
The CI build is currently failing due to recent releases of the Rubocop linter.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -51,4 +51,6 @@ Style/GlobalVars:
     - "lib/webmachine/decision/conneg.rb"
 
 AllCops:
+  NewCops: disable
   SuggestExtensions: false
+  TargetRubyVersion: 2.6
```
Comment (orien) [inline .rubocop.yml:56]: Attempt to make the CI build more stable by limiting changes in future Rubocop releases:

- Don't run new cops: this avoids failures when a new cop is introduced that disagrees with the current style.
- Target the minimum version of Ruby supported by the webmachine. This avoids Rubocop suggestions to use features only available in later versions of Ruby.

File: lib/webmachine/quoted_string.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def quote(str)
 
     # Escapes quotes within a quoted string.
     def escape_quotes(str)
-      str.gsub(/"/, '\\"')
+      str.gsub('"', '\\"')
```
Comment (orien) [inline lib/webmachine/quoted_string.rb:31]: Resolve [Style/RedundantRegexpArgument](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantregexpargument).

File: lib/webmachine/dispatcher/route.rb [hunk 3]
```diff
@@ -37,7 +37,7 @@ def self.rfc3986_percent_decode(value)
           result << if encoded_val.nil?
             s.getch
           else
-            [encoded_val[1..-1]].pack('H*')
+            [encoded_val[1..]].pack('H*')
```
Comment (orien) [inline lib/webmachine/dispatcher/route.rb:40]: Resolve [Style/SlicingWithRange](https://docs.rubocop.org/rubocop/cops_style.html#styleslicingwithrange).

File: lib/webmachine/headers.rb [hunk 4]
```diff
@@ -37,12 +37,12 @@ def self.[](*args)
 
     # Fetch a header
     def [](key)
-      super transform_key(key)
+      super(transform_key(key))
```
Comment (orien) [inline lib/webmachine/headers.rb:40]: Resolve [Style/SuperWithArgsParentheses](https://docs.rubocop.org/rubocop/cops_style.html#stylesuperwithargsparentheses).

**Review by seancribbs — APPROVED** [review]
❤️

**Pull request merged**